llvm/utils/repo/repo.cmake

```diff
@@ -24,10 +24,57 @@ else ()
     set (utils_dir /usr/share/repo)
 endif ()
 
-SET (CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -O0 -fno-exceptions -fno-rtti" CACHE STRING "Default C Flags Debug")
-SET (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -fno-exceptions -fno-rtti" CACHE STRING "Default CXX Flags Debug")
-SET (CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3 -fno-exceptions -fno-rtti" CACHE STRING "Default C Flags Release")
-SET (CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -fno-exceptions -fno-rtti" CACHE STRING "Default CXX Flags Release")
+# The user may specify the libcxxabi_include by giving
+# '-Dlibcxxabi_include:STRING=/path/to/clang/include' on the command line
+if (libcxxabi_include)
+	set (libcxxabi_include "${libcxxabi_include}")
+endif ()
+
+# The user may specify the exception and rtti by giving
+# '-Dexception_rtti:BOOL=Yes' on the command line
+if (exceptions_rtti)
+	set (exceptions_rtti "-fexceptions -frtti")
+else ()
+	set (exceptions_rtti "-fno-exceptions -fno-rtti")
+endif ()
+
+# The user may specify the musl_install by giving
+# '-Dmusl_install:STRING=/path/to/musl/installed/directory' on the command line
+if (musl_install)
+	set (musl_install "${musl_install}")
+else ()
+	set (musl_install "/home/prepo/musl")
+endif ()
+
+# The user may specify the llvm_install by giving
+# '-Dllvm_install:STRING=/path/to/llvm/installed/directory' on the command line
+if (llvm_install)
+	set (llvm_install "${llvm_install}")
+else ()
+	set (llvm_install "/home/prepo/LLVM")
+endif ()
+
+# The user may use the LLVM libc++.a by giving
+# '-Dlibcxx:BOOL=Yes' on the command line
+if (libcxx)
+	set (libcxx "-stdlib=libc++")
+	set (libcxxabi_lib "-stdlib=libc++ -lc++abi")
+endif()
+
+# The user may use the MUSL libc.a by giving
+# '-Dmusl:BOOL=Yes' on the command line
+if (musl)
+	SET (MUSL_COMPILE_FLAGS "-nostdinc -nostdinc++ --sysroot ${musl_install} -isystem ${musl_install}/include ${libcxxabi_include}")
+	SET (CMAKE_EXE_LINKER_FLAGS "-nostdlib -nodefaultlibs -static --sysroot ${musl_install} -L ${musl_install}/lib -L ${llvm_install}/lib ${libcxxabi_lib} -lc" CACHE STRING "toolchain_exelinkflags" FORCE)
+endif()
+
+SET (CMAKE_C_FLAGS "${MUSL_COMPILE_FLAGS} ${exceptions_rtti}" CACHE STRING "toolchain_cflags")
+SET (CMAKE_CXX_FLAGS "${MUSL_COMPILE_FLAGS} ${exceptions_rtti} ${libcxx}" CACHE STRING "toolchain_cxxflags")
```

Comment:
I'm not sure about blasting the values of CMAKE_XX_FLAGS. These are set by the user if they want to override the default compilation flags and this will (I think) silently override the user's choice.

Reply:
I tried to set CMAKE_XX_FLAGS from both command line and repo.cmake file. For example:
```
        cmake -D musl=Yes -D libcxx=Yes                                       \
                -D CMAKE_BUILD_TYPE=Release                                   \
                -D CMAKE_CXX_FLAGS="-frtti -fcxx-exceptions"                  \
                -D LLVM_PATH=../                                              \
                -D LIBUNWIND_ENABLE_SHARED=No                                 \
                -D LIBUNWIND_ENABLE_STATIC=Yes                                \
                -D CMAKE_TOOLCHAIN_FILE=$(TOOLCHAIN_FILE)                     \
                -D CMAKE_INSTALL_PREFIX=$(LLVM)                               \
                ../libunwind &&                                               \
```

I found the `CMAKE_CXX_FLAGS` is set to the "-frtti -fcxx-exceptions" instead of the value in the repo.cmake file. Therefore, the CMAKE_XX_FLAGS value in the repo.cmake file will not override the user's choice in the command line.